packages/react/src/components/Search/Search.js

```diff
@@ -43,6 +43,11 @@ export default class Search extends Component {
      */
     id: PropTypes.string,
 
+    /**
+     * Optional prop to specify the kind of this component
+     */
+    kind: PropTypes.oneOf(['default', 'expandable']),
```

Comment:
One quick question, curious what you think between using `kind` props versus named variants of a component.

e.g. we have `<Search kind="expandable" />` or `<ExpandableSearch />`

I've definitely leaned towards the latter since the `kind` prop makes it harder to group related functionality around a variant but I'm super curious what you think @janhassel

Reply:
Could the prop name also be something like `isExpandable`? 

Or we could mimic the `TableToolbarSearch` prop, and use `persistent` and default that to `true`?

Reply:
I initially thought this variant would not be big enough to make into its own component but after looking further into some other components, maybe the guidance could be to use `kind` whenever the appearance is the main differentiator and creating a new component whenever the behaviour is affected. In that case, this should be an `<ExpandableSearch />` as @joshblack mentioned.

This is based on the following:
| Component | `kind` | wrapper | Effect |
| :- | :- | :- | :- |
| Button | X | | Changes appearance: primary, secondary, ghost, danger |
| Notification | X | | Changes apperance: success, error, warning, information |
| Tile | | X | Changes behaviour: ClickableTile, SelectableTile, RadioTile |
| ContextMenu | | X | Changes behaviour: ContextMenuItem, ContextMenuSelectableItem, ContextMenu |

The (somewhat) recent change in `Toggle` / `ToggleSmall` also reflects this rule: https://github.com/carbon-design-system/carbon/pull/7380

Let me know what you think about this approach and I'll update the PR accordingly.

Reply:
@janhassel that's a great breakdown. When the prop controls _behavior_ (or even something like markup potentially) the wrapper approach works great. If it's a visual change only (like classes) then inlining in the component makes more sense.

Following that chart, does it make sense for `ExpandableSearch` to be a wrapper, or am I misunderstanding? 👀

Reply:
@joshblack Yes, I'd say this should be a wrapper component called `ExpandableSearch` following above's approach. If you agree, @tw15egan then I'll go ahead and update the PR.

Reply:
Let's do it! 👍🏻

Reply:
@joshblack @tw15egan Just pushed an update. I tried syncing the prop types with `ExpandableSearch.propTypes = Search.propTypes` which seems to work fine for validation but the storybook docs don't pick them up. Do you guys have any idea on how to solve this?